### PR TITLE
Oracle/ol support

### PR DESCRIPTION
**Pull request by jemarch**
Hello.

We would like to contribute support for Oracle Linux to crosstool-ng.
This pull-request includes:
a) Support for fetching package versions from git branches and tags.
b) Support for Oracle packages:
    - OL7.9 binutils, gcc and glibc
    - UEK5/u4 kernel
c) Sample configurations for arm-ol7u9-gnueabi and arm-ol7-u9-gnueabihf

Thanks.

### CODE REVIEW

**Review by abrodkin — COMMENTED** [review]
IMHO looks good with just one note: when git is used for fetching sources it does not recognize that needed tarball was already generated and so even if you do `./ct-ng source` before building or on the subsequent re-build you'll see git re-fetching entire repo.

This is at least not elegant, at worst doesn't look like building something "stable"... well and breaks off-line building obviously. That needs to be solved, as in case of use of tags we shouldn't mess with re-fetching anything.

File: packages/linux/package.desc [hunk 1]
```diff
@@ -1,3 +1,4 @@
+origin='The Linux Foundation'
```
[the remark this answers is not in the document]
Comment (jemarch) [reply packages/linux/package.desc:1]: Ok, I will s/The Linux Foundation/www.kernel.org/.  It is certainly more informative :)

Regarding the detection of downloaded/created tarballs, I am taking a look.  Thanks for the comments!

[the remark this answers is not in the document]
Comment (jemarch) [reply packages/linux/package.desc]: All right I looked, and crosstool-ng does not refetch the whole git repo more than once.  What happens on subsequent ct-ng invocations is that the last revision in the specified branches/tags are checked using git ls-remote URL.

We could avoid that by specifying a specific revision, but we would rather avoid that.

[the remark this answers is not in the document]
Comment (abrodkin) [reply packages/linux/package.desc]: Thanks for looking into it! I guess one improvement here might be to check if what was specified is an existing in the current clone tag and then omit `git ls-remote` execution.

[the remark this answers is not in the document]
Comment (jemarch) [reply packages/linux/package.desc]: But that wouldn't work for branches, which may be updated in the remote repo.  ct-ng considers branches and tags as the same thing.

**Comment by jemarch** [comment]
ping

**Review by abrodkin — APPROVED** [review]
Looks good to me. Over to @cpackham

**Comment by cpackham** [comment]
@jemarch this is on my list. I wanted to take a bit of a closer look which I why I haven't just clicked the button on GitHub. Hopefully I get to it over the weekend.

**Comment by jemarch** [comment]
@cpackham  thank you! :)

**Comment by cpackham** [comment]
I've merged this but because I fixed up some whitespace and squashed the packages/linux/package.desc changes github hasn't figured out this PR is closed.

**Comment by jemarch** [comment]
Thank you!

**Pull request closed without merging**